app/services/retrieve_pipeline_viz_graph_data_service.rb

```diff
@@ -176,14 +176,27 @@ def input_output_to_file_paths
     input_output_to_file_paths
   end
 
-  def file_path_to_outputting_step
+  def file_path_to_outputting_step(file_path_to_info)
     file_path_to_outputting_step = {}
     @all_dag_jsons.each_with_index do |stage_dag_json, stage_index|
       stage_dag_json["steps"].each_with_index do |step, step_index|
         stage_dag_json["targets"][step["out"]].each do |file_name|
           file_path = "#{stage_dag_json['output_dir_s3']}/#{@pipeline_run.pipeline_version}/#{file_name}"
           file_path_to_outputting_step[file_path] = { from: { stageIndex: stage_index, stepIndex: step_index } }
         end
+        # Add insert size metrics files to output if available
```

Comment:
This seems to break the abstraction inherent in`file_path_to_outputting_step` by adding specific outputs ad-hoc, is there a way to avoid that with a more elegant design?

Reply:
See my comment above of how to report these additional files to the client. We can make these generic.